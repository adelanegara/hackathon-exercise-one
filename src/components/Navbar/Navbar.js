import React from "react";
import { connect } from "react-redux";
import { AppBar, Box, Toolbar, Typography, Button } from "@mui/material";
import { toast } from "react-toastify";

const Navbar = ({ isLogin, onLogout }) => {


  return (
    <div>
      <Box sx={{ flexGrow: 1 }} data-testid="navbar">
        <AppBar sx={{ bgcolor: "#fee715ff", p: "10px" }} position="static">
          <Toolbar>
            <Typography
              variant="h4"
              component="div"
              sx={{ flexGrow: 1, color: "black" }}
            >
             PARKING BOOKING
            </Typography>
            {isLogin && (
            <Button
              data-testid="button-logout"
              style={{
                textDecoration: "none",
                backgroundColor: "black ",
                padding: "8px 12px",
                color: "white",
              }}
              onClick={onLogout}
            >
              Logout
            </Button>
            )}
          </Toolbar>
        </AppBar>
      </Box>
    </div>
  );
};

const mapStateToProps = (state) => ({
  isLogin: state.isLogin,
});

const mapDispatchToProps = (dispatch) => ({
  onLogout: () => {
    dispatch({ type: "LOGOUT" });
  },
});

export { Navbar as NavbarUnwrapped };
export default connect(mapStateToProps, mapDispatchToProps)(Navbar);
