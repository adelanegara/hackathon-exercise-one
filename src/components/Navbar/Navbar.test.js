import React from "react";
import { screen, render } from "@testing-library/react";
import { NavbarUnwrapped } from "./Navbar";

let wrapper;
beforeEach(() => {
  const { getByTestId } = render(<NavbarUnwrapped />);
  wrapper = getByTestId("navbar");
});

test("test navbar", () => {
  expect(wrapper).toBeInTheDocument();
});

test("Test title", () => {
  const wrapper = screen.getByText(/PARKING BOOKING/);
  expect(wrapper).toBeInTheDocument();
});

test("Test title ABC Bank", () => {
  const wrapper = screen.getByText(/PARKING BOOKING/);
  expect(wrapper).toMatchSnapshot();
});
