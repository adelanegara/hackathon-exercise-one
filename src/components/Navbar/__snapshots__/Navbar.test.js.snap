// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test title ABC Bank 1`] = `
<div
  class="MuiTypography-root MuiTypography-h4 css-scyxek-MuiTypography-root"
>
  PARKING BOOKING
</div>
`;
